### PR TITLE
Add Weights & Biases (W&B) Fabric Logger

### LINKED ISSUE #17885: Support WandBLogger in Fabric
### Description & Motivation

Right now there are only the following loggers available in `lightning.fabric`:

- Logger
- CSVLogger
- TensorBoardLogger

I would like for there to be a `WandbLogger`, similar to that available for the regular `Trainer` interface.

### Alternatives

- It seems I cannot use the regular `WandbLogger`, as it needs to be passed to a `Trainer` object. 
- I tried to use `wandb` directly, but it seems to launch multiple runs for each ddp process.

cc @borda @awaelchli @morganmcg1 @borisdayma @scottire @parambharat

### PR DESCRIPTION
**Pull request by ash0ts**
## What does this PR do?

- Port the logic of the `lightning.pytorch.loggers.WandbLogger` into `lightning.fabric.loggers.WandbLogger`. 
- Added tests to ensure both the current pytorch logger and fabric logger both pass the same test suite as when it was just the pytorch logger
- Added stripped down version of identical tests from the pytorch logger for the fabric logger without any pytorch specific functionality

Fixes #17885 

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [X] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [X] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [X] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

----
:books: Documentation preview :books:: https://pytorch-lightning--19077.org.readthedocs.build/en/19077/

### CODE REVIEW

**Comment by ash0ts** [comment]
I'm having difficulty adding in the resultant docs page for the API on the logger if anyone is able to assist!

**Comment by ash0ts** [comment]
@awaelchli I noticed some failing tests. Would love discuss how I can get these resolved!

**Comment by ash0ts** [comment]
Is there anything I can pointedly fix here? Would love to discuss!

File: src/lightning/fabric/loggers/wandb.py [hunk 1]
```diff
@@ -0,0 +1,701 @@
...
+
+    from lightning.pytorch.callbacks.model_checkpoint import (
+        ModelCheckpoint,  # TODO: Replace with new Fabric Checkpoints system
+    )
+
+_WANDB_AVAILABLE = RequirementCache("wandb>=0.15.0")
+
+
+class WandbLogger(Logger):
+    r"""Log using `Weights and Biases <https://docs.wandb.ai/integrations/lightning>`_.
```
Comment (Borda) [inline src/lightning/fabric/loggers/wandb.py:44]: I think this long docs shall have own page

Comment (ash0ts) [reply src/lightning/fabric/loggers/wandb.py:44]: Hey @Borda! 
I am updating that linked page to show both the lightning and fabric workflows via selectable tabs. If I split the docs I will make sure to change the linked page!

**Comment by ash0ts** [comment]
Hey! @Borda sorry for the bump. Hope your holidays are/were lovely. Would love to discuss what can be fixed to merge this!

**Comment by ash0ts** [comment]
```docstring of lightning.fabric.loggers.wandb.WandbLogger:227:py:class reference target not found: wandb.sdk.wandb_run.Run
```

This specific error is being raised by Sphinx however in my Fabric docstrings I do not refer to `:class: wandb.sdk.wandb_run.Run` outside of TypeChecks

**Review by Borda — APPROVED** [review]

**Comment by awaelchli** [comment]
Closing in favor of #19451 
Great work adding it to wandb as an integration!
